pkg/arvo/app/dojo.hoon

```diff
@@ -193,23 +193,22 @@
       (stag %show (cook $?(%1 %2 %3 %4 %5) (cook lent (stun [1 5] wut))))
     ==
   ::
-  ++  parse-cables
-    %+  cook
-      |=  cables=(list cable:clay)
-      :+  0  %ex
-      ^-  hoon
-      ::
-      :-  %clsg
-      %+  turn  cables
-      |=  cable=cable:clay
-      ^-  hoon
-      ::
-      :+  %clhp
-        ?~  face.cable
-          [%rock %n ~]
-        [%clhp [%rock %n ~] [%sand %tas u.face.cable]]
-      [%sand %tas file-path.cable]
-    (most ;~(plug com gaw) parse-cable)
+  ++  parse-ford
+    |=  base-path=@ta
+    %-  cook  :_  (most ;~(plug com gaw) parse-cable)
+    |=  cables=(list cable:clay)
+    :+  0  %tu
+    ::
+    %+  turn  cables
+    |=  cable=cable:clay
+    ^-  dojo-source
+    ::  TODO: The following *does* properly add faces to imported cores,
+    ::  but discards all their type information, making them hard to
+    ::  work with
```

Comment:
With this version, all imports act like direct/bunt imports (e.g. `/-  *dns`). I was able to use a `%do` command to enable named imports, but my method stripped all type information from the imported files. Any advice/insights on how to fix this would be greatly appreciated!

Reply:
I was able to fix this by using a wet gate, but suggestions on how to clean up this part of the code would still be welcome!

Reply:
I think you’d be better off parsing into just a cable data structure and including the cable itself in Dojo’s state for this session.  You might need to add a new type of dojo-source (or maybe dojo-sink, not sure).

Changing Dojo’s import state should trigger it to send a %warp to Clay to run whatever build is specified by that cable.  The build result will come back as a vase, and you can then use the face in the stored cable data structure to modify the type in the vase to include it in the “dojo subject” — i.e. the subject Dojo uses to run user-specified commands.

A more sophisticated import system would also subscribe to Clay for future versions of this import so the imported file would automatically get rebuilt and loaded into the dojo subject whenever Clay commits to that desk.  I recommend building this as a one-off import command first, though, and only attempting live imports as a second step.  The asynchronicity of live imports makes them more difficult to get right.